src/utils/time-format.js

```diff
@@ -33,15 +33,86 @@ function toHHMMSS(input: number): string {
  */
 function toSecondsFromHHMMSS(input: string): number {
   const parts = input.split(':');
-  let seconds = 0;
   if (parts.length === 2) {
-    if (parseInt(parts[0]) > 59 || parseInt(parts[1]) > 59) {
-      return 0;
-    }
-    seconds += parseInt(parts[0]) * 60;
-    seconds += parseInt(parts[1]);
+    return _toSecondsFromArray(parts, false);
+  }
+  if (parts.length === 3) {
+    return _toSecondsFromArray(parts, true);
   }
+
+  return 0;
+}
+
+/**
+ * Handling the time parts array
+ *
+ * @param {Array<string>} parts the array
+ * @param {boolean} hasHours whether the parts array contains hours or not
+ * @returns {number} number of seconds
+ * @private
+ */
+function _toSecondsFromArray(parts: Array<string>, hasHours: boolean): number {
+  const minutesIndex = hasHours ? 1 : 0;
+  const secondsIndex = minutesIndex + 1;
+  let seconds = 0;
+
+  const minutesPart = parseInt(parts[minutesIndex]);
+  const secondsPart = parseInt(parts[secondsIndex]);
+
+  if (isNaN(minutesPart) || minutesPart > 59 || isNaN(secondsPart) || secondsPart > 59) {
+    return 0;
+  }
+  if (hasHours && !isNaN(parseInt(parts[0]))) {
+    seconds += parseInt(parts[0]) * 3600;
+  }
+  seconds += minutesPart * 60;
+  seconds += secondsPart;
   return seconds;
 }
 
-export {toHHMMSS, toSecondsFromHHMMSS};
+/**
+ * Converting an input that represents seconds to valid time format (hh:mm:ss), i.e. 62 to 01:02
+ *
+ * @param {string} valInSeconds - the value in seconds to format
+ * @returns {string} the formatted value
+ */
+function convertSecondsToTimeFormat(valInSeconds: string): string {
+  const res = new Date(parseInt(valInSeconds) * 1000).toISOString().substr(11, 8);
+  if (res.slice(0, 3) === '00:') {
+    // check if we need to cut the hours part
+    return res.substr(3, 5);
+  }
+  return res;
+}
+
+/**
+ * Formatting input that has only numbers to valid time (hh:mm:ss), i.e. 1 to 00:01
+ *
+ * @param {string} val - the value to format
+ * @returns {string} the formatted value
+ */
+function formatOnlyNumbersInput(val: string): string {
```

Comment:
if it's know that it's only numbers at this point, you can convert to numbers and check
val.length < 10
and
val.length < 59

i think that would be clearer